### DISCUSSION TITLE
Make quimb even more of a generic tensor networks library

### DISCUSSION THREAD

**Discussion by nikitn2** [jcmgray/quimb#251] [Ideas]
Hi Johnnie,

I don't know if you're following iTensor, but it appears they are taking the view there that TNs are a general linear algebra tool that can be employed well beyond quantum physics/computing (like for instance solving PDEs or Machine Learning). To accommodate this, they've added some very nice functionality in their iTensor library:

1. They've included the tensor-cross algorithm: https://arxiv.org/abs/2407.02454
2. Created a DMRG-like solver where a *general linear problem* is solved at the local 1-site/2-sites level instead of just an eigenproblem: https://github.com/ITensor/ITensorTDVP.jl/blob/4e7e55e54e9519183b3eda28ac9e26fc7e587b8a/src/linsolve.jl#L4

Do you think it might be worthwhile to implement this in quimb? It would make it easier to use quimb as a generic TN platform for solving high-dimensional problems.

Would be happy to work together with you on this, if you'd want :) 

Thank you for all your work on quimb – I love it!

**Comment by jcmgray** [comment]
Hi @nikitn2, thanks for raising these!

I'm certainly very pro that perspective and would love to see implementations for both of these in `quimb`.

I don't have any immediate projects that involve either of these things however, so am unlikely to be a major part of implementing them myself any time soon. Of course, it would be great if you wanted to contribute them, and I'd be happy to review and discuss etc. 

Maybe we could start an issue for each to see what features are possible and how an API might look?
